packages/core/src/common/contribution-filter/filter.ts

```diff
@@ -0,0 +1,48 @@
+/********************************************************************************
+ * Copyright (C) 2021 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+export const Filter = Symbol('Filter');
+/**
+ * A `Filter` can be used to test whether a given object should be filtered
+ * from a set of objects. The `test` function can be applied to an object
+ * of matching type and returns `true` if the object should be filtered out.
+ */
+export interface Filter<T extends Object> {
```

Comment:
`Object` is a catch all. It is different from `object`. Almost everything except `null` and `undefined` are `Object`. It allows typing to know about `Object.constructor`.

Reply:
I think `Filters` being an interface can be more convenient since it makes holding state easier if needed. At least it sounds more convenient than closures. Either would work.

Reply:
:shrug: Wat?! ;-)

Reply:
> 
> 
> I think `Filters` being an interface can be more convenient since it makes holding state easier if needed. At least it sounds more convenient than closures. Either would work.

I disagree here philosophically: an interface should pose minimal restrictions on it's implementers. If all we need to do is call a function, that's what we should require.